monai/transforms/utils.py

```diff
@@ -368,6 +369,64 @@ def check_non_lazy_pending_ops(
         warnings.warn(msg)
 
 
+def map_and_generate_sampling_centers(
```

Comment:
Thanks for the PR, would be better to add a unittest include this new function and also mention the size limits for input data, such as whether channel first is required in the docstring.

Thanks

Reply:
Yes this should be added, we can do with a few small tests  for this function since the real work is done by functions tested elsewhere.

Reply:
add test/test_map_and_generate_sampling_centers.py